src/diagnose/patchlib.py

```diff
@@ -107,7 +106,24 @@ def wrapper(*args, **kwargs):
         _resolved_target = primary_patch.getter()
         refs = gc.get_referrers(original)
         for ref in refs:
-            if not isinstance(ref, dict):
+            # exclude frames
+            if isinstance(ref, types.FrameType):
+                continue
+            # exclude cells,
+            if sys.version_info >= (3, 8):
+                if isinstance(ref, types.CellType):
+                    continue
+                # with python < 3.6 types.CellType is not available
+            elif getattr(type(ref), "__name__", None) == "cell" and hasattr(
+                ref, "cell_contents"
+            ):
+                continue
+
+            # with py >= 3.7 the referrer is directly the instance/class object
+            # in this case the patching is applied to its __dict__
+            if not isinstance(ref, dict) and hasattr(ref, "__dict__"):
+                ref = ref.__dict__
+            elif not isinstance(ref, dict):
```

Comment:
This is a lot of isinstance calls, which are expensive. Is there a way to invert this (`if isinstance(ref, dict): ...`) or otherwise simplify it?

Reply:
On a second look both frames and cells do not have `__dict__`, so they're excluded anyway
Check the last commit I added